debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleValueConverters.java

```diff
@@ -569,39 +574,36 @@ private Object resolveTimestampString(Column column, Field fieldDefn, String dat
             else {
                 dateTime = LocalDateTime.from(TIMESTAMP_FORMATTER.parse(dateText.trim()));
             }
-            return getDateTimeWithPrecision(column, dateTime);
+            return dateTime.atZone(GMT_ZONE_ID).toInstant();
         }
 
         final Matcher toDateMatcher = TO_DATE.matcher(data);
         if (toDateMatcher.matches()) {
             dateTime = LocalDateTime.from(TIMESTAMP_FORMATTER.parse(toDateMatcher.group(1)));
-            return getDateTimeWithPrecision(column, dateTime);
+            return dateTime.atZone(GMT_ZONE_ID).toInstant();
         }
 
         // Unable to resolve
         return null;
     }
 
-    private Object getDateTimeWithPrecision(Column column, LocalDateTime dateTime) {
-        if (adaptiveTimePrecisionMode || adaptiveTimeMicrosecondsPrecisionMode) {
-            if (getTimePrecision(column) <= 3) {
-                return dateTime.atZone(GMT_ZONE_ID).toInstant().toEpochMilli();
-            }
-            if (getTimePrecision(column) <= 6) {
-                return Conversions.toEpochMicros(dateTime.atZone(GMT_ZONE_ID).toInstant());
-            }
-            return dateTime.atZone(GMT_ZONE_ID).toInstant().toEpochMilli() * 1_000_000;
-        }
-        return dateTime.atZone(GMT_ZONE_ID).toInstant().toEpochMilli();
-    }
-
     @Override
     protected Object convertTimestampWithZone(Column column, Field fieldDefn, Object data) {
         if (data instanceof String) {
             final Matcher toTimestampTzMatcher = TO_TIMESTAMP_TZ.matcher((String) data);
             if (toTimestampTzMatcher.matches()) {
                 String dateText = toTimestampTzMatcher.group(1);
-                data = ZonedDateTime.from(TIMESTAMP_TZ_FORMATTER.parse(dateText.trim()));
+                ZonedDateTime zonedDateTime = ZonedDateTime.from(TIMESTAMP_TZ_FORMATTER.parse(dateText.trim()));
+                if (OracleTypes.TIMESTAMPLTZ == column.jdbcType()) {
+                    // LogMiner does not supply a TZ and so the converted value needs to account for the database timezone
+                    // relative to the local time zone of the connector.
```

Comment:
The TZ of the connector should not make any difference (that's what we achieved for other connectors after lots of tears, sweat and pain ;).

Instead, this type should be exported based on UTC, as we e.g. do for TIMESTAMP (WITHOUT TZ) for MySQL. In order to do so, you may have to set the connection/session's TZ to UTC. This should also render the separate call to `getDatabaseOffset()` superfluous, as the returned value should [be converted](https://dba.stackexchange.com/questions/159450/can-you-help-clear-confusion-of-oracle-timestamp-with-local-time-zone) to the session's TZ.